include/fmt/printf.h

```diff
@@ -257,7 +257,10 @@ class printf_arg_formatter : public detail::arg_formatter_base<Range> {
         return (*this)(static_cast<int>(value));
       fmt_specs.sign = sign::none;
       fmt_specs.alt = false;
-      fmt_specs.align = align::right;
+      // align::numeric needs to be overwritten here since the '0' flag is
+      // ignored for non-numeric types
```

Comment:
The test which is affected from not overwriting align::numeric here is EXPECT_PRINTF("0000x", "%05c", 'x').